### ISSUE TITLE
Allow for Configuration of Jackson StreamReadConstraints

### ISSUE THREAD

**Issue by jjathman** [micronaut-projects/micronaut-core#9862]
### Feature description

With Jackson 2.15 a new `StreamReadConstraints` [was added](https://github.com/FasterXML/jackson-core/issues/863) which limit the maximum size of certain primitive JSON values. In our application we sometimes receive a String within a JSON message which exceeds the default maximum string length now. Having easy configuration of these stream read constraints would be very helpful for us.

Right now, I think the only way to use something other than the default values is to implement a method like this to override the default JsonFactory created by Micronaut:

```
    @Primary
    @Requires(beans = JacksonConfiguration.class)
    @Singleton
    @BootstrapContextCompatible
    public JsonFactory jsonFactory(JacksonConfiguration jacksonConfiguration) {
        final TSFBuilder<?, ?> jsonFactoryBuilder = JsonFactory.builder();
        StreamReadConstraints streamReadConstraints = StreamReadConstraints.builder()
                .maxStringLength(20 * 1024 * 1024).build()
        jsonFactoryBuilder.streamReadConstraints(streamReadConstraints)
        jacksonConfiguration.getFactorySettings().forEach(jsonFactoryBuilder::configure);
        return jsonFactoryBuilder.build();
    }
```

**Comment by jjathman** [comment]
Note, the default was changed in a subsequent patch release which does fix the immediate problem for us, but I still think this would be nice to make configurable. https://github.com/FasterXML/jackson-core/issues/1014

**Comment by yawkat** [comment]
note that the micronaut-serialization constraints are already configurable through `micronaut.serde.maximum-nesting-depth`

(doesn't help when you're using jackson-databind ofc)

**Comment by ledigiacomo** [comment]
@jjathman is https://github.com/FasterXML/jackson-core/pull/1019 which was made available in version 2.15.2 what you were looking for?

**Comment by jjathman** [comment]
@ledigiacomo I don't think that setting is configurable is using Micronaut's built in Jackson support is it? Maybe I don't understand how that would be used.

**Comment by graemerocher** [comment]
Triage: this remains a useful Jackson integration enhancement. Core's `JacksonConfiguration` exposes feature maps and custom-factory hooks, but no direct `StreamReadConstraints` settings; the current source has no `StreamReadConstraints` support. Add nullable/configurable limits for max string length, number length, nesting depth, and document length, apply them to the default `JsonFactory` builder before custom factory settings, and preserve the existing `@Primary JsonFactory` override path. Tests should prove a long string fails or succeeds according to `jackson.stream-read-constraints.*` configuration.
